catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/BasicTypes.java

```diff
@@ -311,7 +328,36 @@ public AttributeFormat getAttributeFormat() {
                 false /* tokenized */,
                 true /* multivalued */,
                 STRING_TYPE));
-        return descriptors;
+        DESCRIPTORS = Collections.unmodifiableSet(descriptors);
+
+        BASIC_METACARD = new MetacardTypeImpl(MetacardType.DEFAULT_METACARD_TYPE_NAME,
```

Comment:
From an OO standpoint, the implication of this API is that it is an immutable object, since it has no setters. The API is actually that it return an unmodifiable set, see: https://github.com/codice/ddf/blob/bb7177ea3dd800f5d267f8bf41070e704fc9b2bb/catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/MetacardType.java#L41-L41

All this PR is doing is correctly enforcing API constraints that were missed in the initial implementation. I don't think there is anything misleading about this design, it is perfectly consistent with the javadocs.